redbot/cogs/warnings/warnings.py

```diff
@@ -76,6 +84,59 @@ def __init__(self, bot: Red):
         else:
             await ctx.send(_("Custom reasons have been disabled."))
 
+    @warningset.command()
+    @commands.guild_only()
+    async def toggledm(self, ctx: commands.Context):
+        """Toggle whether warnings should be sent to users in DMs."""
+        guild = ctx.guild
+        toggle = not await self.config.guild(guild).toggle_dm()
+        await self.config.guild(guild).toggle_dm.set(toggle)
+        if toggle:
+            await ctx.send(_("I will now try to send warnings to users DMs."))
+        else:
+            await ctx.send(_("Warnings will no longer be sent to users DMs."))
+
+    @warningset.command()
+    @commands.guild_only()
+    async def warnchannel(self, ctx: commands.Context, channel: discord.TextChannel = None):
+        """Set the channel where warnings should be sent to.
+
+        Leave empty to use the channel `[p]warn` command was called in.
+        """
+        guild = ctx.guild
+        if channel:
+            await self.config.guild(guild).warn_channel.set(channel.id)
+            await ctx.send(
+                _("The warn channel has been set to {channel}.").format(channel=channel.mention)
+            )
+        else:
+            await self.config.guild(guild).warn_channel.set(channel)
+            await ctx.send(_("Warnings will now be sent in the channel command was used in."))
+
+    @warningset.command()
+    @commands.guild_only()
+    async def togglechannel(self, ctx: commands.Context):
+        """
+        Toggle if warnings should be sent to a channel set with `[p]warningset warnchannel`.
+        """
+        guild = ctx.guild
+        toggle = await self.config.guild(guild).toggle_channel()
+        await self.config.guild(guild).toggle_channel.set(not toggle)
+        channel = self.bot.get_channel(await self.config.guild(guild).warn_channel())
+        if not toggle:
+            if channel:
+                await ctx.send(
+                    _("Warnings will now be sent to {channel}.").format(channel=channel.mention)
+                )
+            else:
+                await ctx.send(
+                    _("Toggle channel has been enabled but a channel hasn't been set.").format(
```

Comment:
This should say that warnings will now be sent to the channel command was used in, instead of saying that channel hasn't been set.
And no need to use `.format` if there's nothing to be replaced.